aiogram/utils/executor.py

```diff
@@ -69,9 +69,9 @@ def set_webhook(dispatcher: Dispatcher, webhook_path: str, *, loop: Optional[asy
     return executor
 
 
-def start_webhook(dispatcher, webhook_path, *, loop=None, skip_updates=None,
-                  on_startup=None, on_shutdown=None, check_ip=False, retry_after=None, route_name=DEFAULT_ROUTE_NAME,
-                  **kwargs):
+def start_webhook(dispatcher, webhook_path, *, loop=None, skip_updates=None, on_startup=None, on_shutdown=None,
+                  check_ip=False, retry_after=None, route_name=DEFAULT_ROUTE_NAME,
+                  custom_routes: List[List[Union[str, str, Coroutine[Any, Any, Response]]]] = None, **kwargs):
```

Comment:
strange type, don't you find?

Reply:
It seems to me, this is the correct type-hinting for the nested list of lists of the union: method (str), path (str), and handler, isn't it?

Reply:
This type-hinting can be removed, as you wish.

Reply:
it's about correctness of your type-hint.

`List[List[Union[str, str, Coroutine[Any, Any, Response]]]]`

- First thing to look at is `Union[str, str]` which always gives `str`. [docs](https://docs.python.org/3/library/typing.html#typing.Union)

- Secondly, `var = List[Union[str, str, X]]` narrows to `var = List[Union[str, X]]` and what do we get from the type-hint now? that `var` is a list of either string or X or both. but if we want...
> method (str), path (str), and handler
we probably want to use `List[Tuple[X...]]` instead of `List[List[Union[X...]]]`

- Lastly, why are we waiting for `Coroutine[Any, Any, Response]` for `handler` param?
I'm sure it's not what aiohttp really expects. What it really does expect from you is `Callable[[Request], Awaitable[StreamResponse]]` or `AbstractView`

Let me know, if anything was unclear or incorrect in my comment. typing is sure a great topic to discuss :)

Reply:
Oh, you're completely right.
I realized that I was using Union wrong.
What about the last remark, actually it used to be something like this:

> Callable[[Request], Awaitable[StreamResponse]] or AbstractView

but PyCharm showed a warning like _expected ..., got `Coroutine[Any, Any, Response]` instead._

Thanks for your critics, since I'm a beginner I really appreciate it.